R/check.R

```diff
@@ -1,91 +1,80 @@
-    rlang::abort(paste0("argument ", name, " is not a epirange"),
-      name = name, value = value, class = "invalid_argument"
-    )
-  }
+#' Allows a vector of numeric values or a list of numeric values
+#' @importFrom checkmate assert_integerish
+check_integerish_param <- function(name, value, required = TRUE) {
+  null.ok <- !required
+  assert_integerish(value, null.ok = null.ok, any.missing = FALSE, .var.name = name)
 }
 
-check_epirange_param <- function(name, value, required = TRUE) {
-  if (is.null(value)) {
-    if (required) {
-      rlang::abort(paste0("argument ", name, " is not a epirange"),
-        name = name, value = value, class = "invalid_argument"
-      )
-    }
-    return()
-  }
-  if (is_epirange_like(value)) {
-    return()
-  }
-  if (!is.list || !all(sapply(sapply(value, is_epirange_like)))) {
-    rlang::abort(paste0("argument ", name, " is not a epirange"),
-      name = name, value = value, class = "invalid_argument"
-    )
-  }
+#' Allows a scalar date_like param: a single date, character, or integerish
+#' @importFrom checkmate assert check_date check_character check_integerish
+check_scalar_date_param <- function(name, value, required = TRUE) {
+  null.ok <- !required
+  assert(
+    # check_date(value, null.ok = null.ok, len = 1),
+    check_character(value, null.ok = null.ok, len = 1),
+    check_integerish(value, null.ok = null.ok, len = 1),
+    combine = "or",
+    .var.name = name
+  )
 }
 
-check_single_string_param <- function(name, value, required = TRUE) {
-  if ((required &&
-    !(!is.null(value) &&
-      is.character(value) &&
-      length(value) == 1)) ||
-    (!required &&
-      !(is.null(value) ||
-        (
-          is.character(value) && length(value) == 1
-        )))) {
-    rlang::abort(paste0("argument ", name, " is not a single string"),
-      name = name, value = value, class = "invalid_argument"
-    )
-  }
+#' Allows a vector of date_like params: dates, characters, or integerishes
+#' @importFrom checkmate check_date check_character check_integerish
+check_date_param <- function(name, value, required = TRUE) {
+  null.ok <- !required
+  assert(
+    # check_date(value, null.ok = null.ok),
+    check_character(value, null.ok = null.ok),
+    check_integerish(value, null.ok = null.ok),
+    combine = "or",
+    .var.name = name
+  )
 }
 
-check_single_int_param <- function(name, value, required = TRUE) {
-  if ((required &&
-    !(!is.null(value) &&
-      is.numeric(value) &&
-      length(value) == 1)) ||
-    (!required &&
-      !(is.null(value) || (is.numeric(value) && length(value) == 1)))) {
-    rlang::abort(paste0("argument ", name, " is not a single integer"),
-      name = name, value = value, class = "invalid_argument"
-    )
-  }
+#' Allows a scalar timeset param: a character, an integerish, or an EpiRange
+#' @importFrom checkmate assert check_character check_integerish check_class
+check_scalar_timeset_param <- function(name, value, required = TRUE) {
```

Comment:
Can we call these `assert_*` or whatever the checkmate equivalent is?  checkmate uses `check_*` to indicate the output is TRUE/FALSE.